lib/app/dispatcher.rb

```diff
@@ -68,7 +68,11 @@ def start_polling_jobs!
         scheduler.when_triggered do |connector_settings, task|
           case task
           when :sync
-            start_sync_task(connector_settings)
+            # update connector metadata
+            # @TODO rename #claim_job because it doesn't claim a job
+            Core::ElasticConnectorActions.claim_job(connector_settings.id)
```

Comment:
I'm introducing some changes in PR https://github.com/elastic/connectors-ruby/pull/428.

`Core::ElasticConnectorActions` is a wrapper to access ES. IMO, it shouldn't contain any business logic, e.g. `claim_job`, `complete_sync`. It should only contain methods like `update_job`, `create_job`.

I didn't touch `claim_job`, as I think this is more related to the decoupling work. I will eventually remove `complete_sync` method. Whenever a job completes, it should call `job.done!` or `job.error!`, where the `job` is a job entity.

Reply:
> Core::ElasticConnectorActions is a wrapper to access ES

What exactly it should be wrapping from your point of view?

Reply:
When I originally introduced `ElasticConnectorActions` its goal was to have a "contract in actions". What I mean by that:

Connectors protocol defines "entities" - Jobs, Connectors - and their properties. However, it's only a part of the protocol, another part of protocol is "actions": e.g. "start a sync", "complete a sync", "update connector configurable fields" - atomic actions that make the protocol actually work. It was done so that it was easier to see how the code inside the framework changes the state of the system - instead of freeform "es_client.update" it does for example:

```
job_response = ElasticConnectorActions.create_job(connector.id)

job_claim_response = ElasticConnectorActions.claim_job(connector.id)

if job_claim_response['success']
  # do stuff
  ElasticConnectorActions.update_running_job_state(job.id, state: RUNNING, indexed_doc_count: 55, deleted_doc_count: 15)

  # finish
  ElasticConnectorActions.finish_job_successfully(job.id)
end
```

However with time this class started to have more generic methods like "update_connector" and "update job" that made the original approach not work that well any more.

So TLDR: this class was supposed to encapsulate connectors protocol "actions", but not provide any domain logic inside at all. It should have just wrapped Elasticsearch client with more "smart" methods.

Reply:
Got it. I think we can label this discussion as tech-debt